From b2d74eed58954fb0e241b0fd573b5b05e632ad31 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Mon, 7 Apr 2014 12:20:48 +0800
Subject: [PATCH 04/14] Update cogl.symbols and cogl-path.symbols

In Lionel's work for supporting introspection better for Cogl, a number of
public symbols were added for Cogl and Cogl-Path, so add these symbols to
cogl.symbols and cogl-path.symbols so that they can be exported, which will
fix the build of the Cogl conformance test and the introspection files
for the Windows-based builds.

Reviewed-by: Neil Roberts <neil@linux.intel.com>
---
 cogl-path/cogl-path.symbols |   3 ++
 cogl/cogl.symbols           | 127 +++++++++++++++++++++++++++++++++++++++++---
 2 files changed, 124 insertions(+), 6 deletions(-)

diff --git a/cogl-path/cogl-path.symbols b/cogl-path/cogl-path.symbols
index fd56795..b643ec0 100644
--- a/cogl-path/cogl-path.symbols
+++ b/cogl-path/cogl-path.symbols
@@ -11,6 +11,9 @@ cogl_path_ellipse
 cogl_path_fill
 cogl_path_fill_preserve
 cogl_path_get_fill_rule
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_path_get_gtype
+#endif
 cogl_path_line
 cogl_path_line_to
 cogl_path_move_to
diff --git a/cogl/cogl.symbols b/cogl/cogl.symbols
index d1eb319..c48314a 100644
--- a/cogl/cogl.symbols
+++ b/cogl/cogl.symbols
@@ -4,14 +4,25 @@
 cogl_android_set_native_window
 #endif
 
-/* cogl-atlas-texture.h */
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_atlas_texture_get_gtype
+#endif
 cogl_atlas_texture_new_with_size
 cogl_atlas_texture_new_from_file
 cogl_atlas_texture_new_from_data
 cogl_atlas_texture_new_from_bitmap
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_attribute_buffer_get_gtype
+#endif
 cogl_attribute_buffer_new_with_size
 
+cogl_attribute_buffer_new
+cogl_attribute_get_buffer
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_attribute_get_gtype
+#endif
+cogl_attribute_get_normalized
 cogl_attribute_new
 cogl_attribute_new_const_1f
 cogl_attribute_new_const_2f
@@ -23,9 +34,6 @@ cogl_attribute_new_const_3x3fv
 cogl_attribute_new_const_4f
 cogl_attribute_new_const_4fv
 cogl_attribute_new_const_4x4fv
-cogl_attribute_buffer_new
-cogl_attribute_get_buffer
-cogl_attribute_get_normalized
 cogl_attribute_set_buffer
 cogl_attribute_set_normalized
 cogl_attribute_type_get_type
@@ -35,6 +43,9 @@ cogl_begin_gl
 cogl_bitmap_error_get_type
 cogl_bitmap_get_buffer
 cogl_bitmap_get_format
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_bitmap_get_gtype
+#endif
 cogl_bitmap_get_height
 cogl_bitmap_get_rowstride
 cogl_bitmap_get_size_from_file
@@ -111,6 +122,9 @@ cogl_color_get_blue_float
 cogl_color_get_green
 cogl_color_get_green_byte
 cogl_color_get_green_float
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_color_get_gtype
+#endif
 cogl_color_get_red
 cogl_color_get_red_byte
 cogl_color_get_red_float
@@ -149,6 +163,9 @@ cogl_glx_context_get_glx_context
 #endif
 
 cogl_context_get_display
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_context_get_gtype
+#endif
 cogl_context_get_renderer
 cogl_context_new
 
@@ -172,6 +189,9 @@ cogl_depth_test_function_get_type
 
 cogl_disable_fog
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_display_get_gtype
+#endif
 cogl_display_get_renderer
 cogl_display_new
 cogl_display_setup
@@ -188,6 +208,9 @@ cogl_error_matches
 cogl_euler_copy
 cogl_euler_equal
 cogl_euler_free
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_euler_get_gtype
+#endif
 cogl_euler_init
 cogl_euler_init_from_matrix
 #if 0
@@ -239,6 +262,9 @@ cogl_framebuffer_get_depth_texture_enabled
 cogl_framebuffer_get_depth_write_enabled
 cogl_framebuffer_get_dither_enabled
 cogl_framebuffer_get_green_bits
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_framebuffer_get_gtype
+#endif
 cogl_framebuffer_get_height
 cogl_framebuffer_get_modelview_matrix
 cogl_framebuffer_get_projection_matrix
@@ -284,7 +310,14 @@ cogl_framebuffer_translate
 cogl_framebuffer_vdraw_attributes
 /* cogl_framebuffer_vdraw_indexed_attributes */ /* Not Implemented! */
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_frame_closure_get_gtype
+#endif
 cogl_frame_info_get_frame_counter
+
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_frame_info_get_gtype
+#endif
 cogl_frame_info_get_output
 cogl_frame_info_get_presentation_time
 cogl_frame_info_get_refresh_rate
@@ -311,6 +344,9 @@ cogl_get_static_identity_quaternion
 cogl_get_static_zero_quaternion
 cogl_get_viewport
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_gles2_context_get_gtype
+#endif
 cogl_gles2_context_get_vtable
 cogl_gles2_context_new
 cogl_gles2_get_current_vtable
@@ -322,8 +358,8 @@ cogl_glib_renderer_source_new
 cogl_glib_source_new
 #endif
 
-#ifdef _COGL_SUPPORTS_GTYPE_INTEGRATION
-/* cogl_gtype_matrix_get_type */ /* Not Implemented */
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_gtype_matrix_get_type
 #endif
 
 cogl_handle_get_type
@@ -333,8 +369,14 @@ cogl_handle_unref
 cogl_has_feature
 cogl_has_features
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_index_buffer_get_gtype
+#endif
 cogl_index_buffer_new
 cogl_indices_get_buffer
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_indices_get_gtype
+#endif
 cogl_indices_get_offset
 cogl_indices_get_type
 cogl_indices_new
@@ -448,6 +490,9 @@ cogl_matrix_copy
 cogl_matrix_entry_calculate_translation
 cogl_matrix_entry_equal
 cogl_matrix_entry_get
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_matrix_entry_get_gtype
+#endif
 cogl_matrix_entry_is_identity
 cogl_matrix_entry_ref
 cogl_matrix_entry_unref
@@ -455,6 +500,9 @@ cogl_matrix_equal
 cogl_matrix_free
 cogl_matrix_frustum
 cogl_matrix_get_array
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_matrix_get_gtype
+#endif
 cogl_matrix_get_inverse
 cogl_matrix_init_from_array
 cogl_matrix_init_translation
@@ -481,6 +529,9 @@ cogl_matrix_scale
 cogl_matrix_stack_frustum
 cogl_matrix_stack_get
 cogl_matrix_stack_get_entry
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_matrix_stack_get_gtype
+#endif
 cogl_matrix_stack_get_inverse
 cogl_matrix_stack_load_identity
 cogl_matrix_stack_multiply
@@ -504,11 +555,17 @@ cogl_matrix_view_2d_in_perspective
 
 cogl_meta_texture_foreach_in_region
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_object_get_gtype
+#endif
 cogl_object_get_user_data
 cogl_object_ref
 cogl_object_set_user_data
 cogl_object_unref
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_offscreen_get_gtype
+#endif
 cogl_offscreen_new_to_texture
 cogl_offscreen_new_with_texture
 
@@ -519,8 +576,14 @@ cogl_onscreen_add_swap_buffers_callback
 #ifndef COGL_WINSYS_INTEGRATED
 cogl_onscreen_clutter_backend_set_size_CLUTTER
 #endif
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_onscreen_dirty_closure_get_gtype
+#endif
 cogl_onscreen_get_buffer_age
 cogl_onscreen_get_frame_counter
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_onscreen_get_gtype
+#endif
 cogl_onscreen_get_resizable
 cogl_onscreen_hide
 cogl_onscreen_new
@@ -529,18 +592,27 @@ cogl_onscreen_remove_dirty_callback
 cogl_onscreen_remove_frame_callback
 cogl_onscreen_remove_resize_callback
 cogl_onscreen_remove_swap_buffers_callback
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_onscreen_resize_closure_get_gtype
+#endif
 cogl_onscreen_set_resizable
 cogl_onscreen_set_swap_throttled
 cogl_onscreen_show
 cogl_onscreen_swap_buffers
 cogl_onscreen_swap_buffers_with_damage
 cogl_onscreen_swap_region
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_onscreen_template_get_gtype
+#endif
 cogl_onscreen_template_new
 cogl_onscreen_template_set_samples_per_pixel
 cogl_onscreen_template_set_swap_throttled
 
 cogl_ortho
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_output_get_gtype
+#endif
 cogl_output_get_height
 cogl_output_get_mm_height
 cogl_output_get_mm_width
@@ -566,6 +638,9 @@ cogl_pipeline_get_depth_state
 cogl_pipeline_get_diffuse
 cogl_pipeline_get_emission
 cogl_pipeline_get_front_face_winding
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_pipeline_get_gtype
+#endif
 cogl_pipeline_get_layer_mag_filter
 cogl_pipeline_get_layer_min_filter
 cogl_pipeline_get_layer_point_sprite_coords_enabled
@@ -618,6 +693,9 @@ cogl_pipeline_set_uniform_1f
 cogl_pipeline_set_uniform_1i
 cogl_pipeline_set_user_program
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_pixel_buffer_get_gtype
+#endif
 cogl_pixel_buffer_new
 #if 0
 /* not exported in the main APIs for now */
@@ -641,6 +719,9 @@ cogl_pop_source
 cogl_primitive_copy
 cogl_primitive_foreach_attribute
 cogl_primitive_get_first_vertex
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_primitive_get_gtype
+#endif
 cogl_primitive_get_indices
 cogl_primitive_get_mode
 cogl_primitive_get_n_vertices
@@ -701,6 +782,9 @@ cogl_quaternion_copy
 cogl_quaternion_dot_product
 cogl_quaternion_equal
 cogl_quaternion_free
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_quaternion_get_gtype
+#endif
 cogl_quaternion_get_rotation_angle
 cogl_quaternion_get_rotation_axis
 cogl_quaternion_init
@@ -733,6 +817,9 @@ cogl_renderer_check_onscreen_template
 cogl_renderer_connect
 cogl_renderer_foreach_output
 cogl_renderer_get_driver
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_renderer_get_gtype
+#endif
 cogl_renderer_get_n_fragment_texture_units
 cogl_renderer_error_get_type
 cogl_renderer_get_winsys_id
@@ -793,6 +880,9 @@ cogl_snippet_get_hook
 cogl_snippet_get_post
 cogl_snippet_get_pre
 cogl_snippet_get_replace
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_snippet_get_gtype
+#endif
 cogl_snippet_new
 cogl_snippet_set_declarations
 cogl_snippet_set_post
@@ -801,9 +891,15 @@ cogl_snippet_set_replace
 
 cogl_sqrti
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_sub_texture_get_gtype
+#endif
 cogl_sub_texture_get_parent
 cogl_sub_texture_new
 
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_swap_chain_get_gtype
+#endif
 cogl_swap_chain_new
 cogl_swap_chain_set_has_alpha
 cogl_swap_chain_set_length
@@ -811,12 +907,16 @@ cogl_swap_chain_set_length
 cogl_system_error_get_type
 
 cogl_texture_allocate
+cogl_texture_components_get_type
 cogl_texture_error_get_type
 cogl_texture_flags_get_type
 cogl_texture_get_components
 cogl_texture_get_data
 cogl_texture_get_format
 cogl_texture_get_gl_texture
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_get_gtype
+#endif
 cogl_texture_get_height
 cogl_texture_get_max_waste
 cogl_texture_get_premultiplied
@@ -831,11 +931,17 @@ cogl_texture_new_from_sub_texture
 cogl_texture_new_with_size
 #ifdef COGL_HAS_X11
 cogl_texture_pixmap_x11_error_domain
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_pixmap_x11_get_gtype
+#endif
 cogl_texture_pixmap_x11_is_using_tfp_extension
 cogl_texture_pixmap_x11_new
 cogl_texture_pixmap_x11_set_damage_object
 cogl_texture_pixmap_x11_update_area
 #endif
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_rectangle_get_gtype
+#endif
 cogl_texture_rectangle_new_from_bitmap
 cogl_texture_rectangle_new_from_foreign
 cogl_texture_rectangle_new_with_size
@@ -851,14 +957,23 @@ cogl_texture_type_get_type
 #ifndef COGL_DISABLE_DEPRECATED
 cogl_texture_unref
 #endif
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_2d_get_gtype
+#endif
 cogl_texture_2d_new_from_bitmap
 cogl_texture_2d_new_from_data
 cogl_texture_2d_new_from_file
 cogl_texture_2d_new_with_size
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_2d_sliced_get_gtype
+#endif
 cogl_texture_2d_sliced_new_from_bitmap
 cogl_texture_2d_sliced_new_from_data
 cogl_texture_2d_sliced_new_from_file
 cogl_texture_2d_sliced_new_with_size
+#ifdef COGL_HAS_GTYPE_SUPPORT
+cogl_texture_3d_get_gtype
+#endif
 cogl_texture_3d_new_from_bitmap
 cogl_texture_3d_new_from_data
 cogl_texture_3d_new_with_size
-- 
2.0.0

